src/components/App/App.js

```diff
@@ -0,0 +1,317 @@
+import MenuPopup from '../MenuPopup/MenuPopup';
+import * as NewsApi from '../../utils/NewsApi';
+import * as MainApi from '../../utils/MainApi';
+import SavedNewsHeader from '../SavedNewsHeader/SavedNewsHeader';
+import SavedNews from '../SavedNews/SavedNews';
+import ProtectedRoute from '../ProtectedRoute';
+
+function App() {
+  const navigate = useNavigate();
+  const [currentUser, setCurrentUser] = useState({});
+  const [token, setToken] = useState(localStorage.getItem('jwt'));
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [loggedInSavedNews, setLoggedInSavedNews] = useState(false);
+  const [savedNews, setSavedNews] = useState([]);
+  const [isLoginPopupOpen, setIsLoginPopupOpen] = useState(false);
+  const [isRegisterPopupOpen, setIsRegisterPopupOpen] = useState(false);
+  const [isInfoTooltipPopupOpen, setIsInfoTooltipPopupOpen] = useState(false);
+  const [isMenuPopupOpen, setIsMenuPopupOpen] = useState(false);
+  const [searchOpen, setSearchOpen] = useState(false);
+  const [preloadOpen, setPreloadOpen] = useState(false);
+  const [nothingFoundOpen, setNothingFoundOpen] = useState(false);
+  const [cards, setCards] = useState(
+    JSON.parse(localStorage.getItem('cards') || '[]')
+  );
+  const [messege, setMessege] = useState(false);
+  const [homeActive, setHomeActive] = useState(true);
+  const [keyWord, setKeyWord] = useState('');
+  const [emailConflict, setEmailConflict] = useState(false);
+
+  React.useEffect(() => {
+    if (token) {
+      MainApi.getUserInfo(token)
+        .then((res) => {
+          setLoggedIn(true);
+          setCurrentUser(res.data);
+        })
+        .catch((err) => console.log(err));
+    } else {
+      setLoggedIn(false);
+    }
+  }, [token]);
+  function handleSavedArticlesClick() {
+    MainApi.getAllArticles(token)
+      .then((res) => {
+        setSavedNews(res);
+        setLoggedInSavedNews(true);
+        setHomeActive(false);
+        closeAllPopups();
+        navigate('/saved-news');
+      })
+      .catch((err) => console.log(err));
+  }
+  function handleHomeClick() {
+    setLoggedInSavedNews(false);
+    setHomeActive(true);
+    closeAllPopups();
+  }
+  function handleSearchClick(keyWord) {
+    setKeyWord(keyWord);
+    setPreloadOpen(true);
+    setSearchOpen(false);
+    setNothingFoundOpen(false);
+    NewsApi.getInitialCards(keyWord)
+      .then((cards) => {
+        setPreloadOpen(false);
+        if (cards.articles.length > 0) {
+          setCards(cards.articles);
+          cards.articles.map((obj) => (obj.saved = 'false'));
+          setSearchOpen(true);
+          localStorage.setItem('cards', JSON.stringify(cards));
+        } else {
+          setNothingFoundOpen(true);
+        }
+      })
+      .catch((err) => {
+        console.log(err);
+        setPreloadOpen(false);
+        setSearchOpen(false);
+        setMessege(true);
+        setIsInfoTooltipPopupOpen(true);
+      });
+  }
+  function deleteArticle(card) {
```

Comment:
* When I try to delete artilces from `Saved articles` the app crashes. See the errors http://joxi.ru/gmvXnBgCejdPyA

Reply:
done